### DISCUSSION TITLE
Higlight "flickering": There is a delay for detecting the right highlight group

### DISCUSSION THREAD

**Discussion by Mickx971** [olimorris/onedarkpro.nvim#248] [Q&A] [answered]
Hello,

First of all I want to thank you for this awesome plugin, highly customizable !

I am currently working on a typescript project and I encounter some highlight issue.

When typing the default color is usually red but for a fraction of second. Then, the correct highlight is displayed according to the syntax.

This is a bit annoying for people with visual sensitivity.

Is there a configuration to answer this issue ?

I do like the theme colors and don't want to change it for a more bearable flickering.

**Comment by olimorris** [comment, marked as answer]
Hey thank you and glad you like it.

I would advise turning on caching in the plugin if it's not already on. That speeds things up dramatically as all of the highlight groups are [cached](https://github.com/olimorris/onedarkpro.nvim/blob/b542d3a18b99d49161b007c9f90530101ff8bedc/lua/onedarkpro/lib/compile.lua#L153) ready to be executed when the plugin loads.

My hunch is that this is Tree-sitter related. Perhaps turn off Typescript filetype highlights in the theme as per the [README](https://github.com/olimorris/onedarkpro.nvim?tab=readme-ov-file#configuring-filetype-highlighting).
